### ISSUE TITLE
ansible_architecture access in map is incorrect

### ISSUE THREAD

**Issue by jack-evans** [andrewrothstein/ansible-protoc#1]
ansible install details:
```bash
$ ansible --version
ansible [core 2.13.4]
  config file = None
  configured module search path = ['/Users/jackevans/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/homebrew/Cellar/ansible/6.4.0/libexec/lib/python3.10/site-packages/ansible
  ansible collection location = /Users/jackevans/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/homebrew/bin/ansible
  python version = 3.10.6 (main, Aug 30 2022, 04:58:14) [Clang 13.1.6 (clang-1316.0.21.2.5)]
  jinja version = 3.1.2
  libyaml = True
```

Trying to use this ansible role to install protoc on a m1 mac and am getting the following error:

```text
TASK [andrewrothstein.protoc : downloading {{ protoc_url }} to {{ protoc_tmp_zip }}] *********************************************************************************************************************************
fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: {{ protoc_mirror }}/v{{ protoc_ver }}/{{ protoc_zip }}: protoc-{{ protoc_ver }}-{{ protoc_platform }}.zip: {{ protoc_os }}-{{ protoc_arch }}: {{ protoc_arch_map[ansible_architecture] }}: 'dict object' has no attribute 'arm64'\n\nThe error appears to be in '/Users/jackevans/.ansible/roles/andrewrothstein.protoc/tasks/main.yml': line 22, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n  block:\n    - name: downloading {{ protoc_url }} to {{ protoc_tmp_zip }}\n      ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes. Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"}

TASK [andrewrothstein.protoc : rm {{ protoc_tmp_zip }}] **************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: /tmp/{{ protoc_zip }}: protoc-{{ protoc_ver }}-{{ protoc_platform }}.zip: {{ protoc_os }}-{{ protoc_arch }}: {{ protoc_arch_map[ansible_architecture] }}: 'dict object' has no attribute 'arm64'\n\nThe error appears to be in '/Users/jackevans/.ansible/roles/andrewrothstein.protoc/tasks/main.yml': line 46, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n  always:\n    - name: rm {{ protoc_tmp_zip }}\n      ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes. Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"}
```

seems like it is trying to access `protoc_arch_map[ansible_architecture]` but is getting empty. On an m1 mac, the `ansible_architecture` is coming back as `arm64` which doesnt match anything defined in the map:

```yaml
protoc_arch_map:
  aarch64: aarch_64
  ppc64le: ppcle_64
  s390x: s390x
  x86_64: x86_64
  x86_32: x86_64
```

I also notice there are a few other modules developed that have a similar issue. happy to look at putting in a PR to fix this if it helps 🙂

**Comment by jack-evans** [comment]
it is possible to get around by overriding in vars:

```yaml
vars:
  protoc_arch_map:
    arm64: aarch_64
```

**Comment by andrewrothstein** [comment]
thanks for letting me know, I'll update the arch_map. published the fix to the galaxy as v1.1.1.
